### ISSUE TITLE
Add a typescript definition file

### ISSUE THREAD

**Issue by abidon** [TooTallNate/node-https-proxy-agent#27]
Using this library in typescript is impossible (with noImplicitAny option enabled) as there is no definition file.

Would you mind adding one ?

**Comment by yonilerner** [comment]
FWIW, I just created an `https-proxy-agent.d.ts` file in my project with:
```typescript
declare module 'https-proxy-agent' {
    import * as https from 'https'

    namespace HttpsProxyAgent {
        interface HttpsProxyAgentOptions {
            host: string
            port: number
            secureProxy?: boolean
            headers: {
                [key: string]: string
            }

            [key: string]: any
        }
    }
    
    // HttpsProxyAgent doesnt *actually* extend https.Agent, but for my purposes I want it to pretend that it does
    class HttpsProxyAgent extends https.Agent {
        constructor(opts: HttpsProxyAgent.HttpsProxyAgentOptions)
    }

    export = HttpsProxyAgent
}
```

And I use
```typescript
import * as HttpsProxyAgent from 'https-proxy-agent'
import {HttpsProxyAgentOptions} from 'https-proxy-agent'

// HttpsProxyAgent.HttpsProxyAgentOptions would also work
const opts: HttpsProxyAgentOptions = {...}

const agent = new HttpsProxyAgent(opts)
```

**Comment by acecilia** [comment]
@yonilerner are you planning to make a new release with this change anytime soon?

**Comment by yesworld** [comment]
@yonilerner Thank you man!

**Comment by TooTallNate** [comment]
Fixed by #66.

**Comment by MichaelHindley** [comment]
@TooTallNate 
Just a heads up, using the index.d.ts from master to work with the currently published 2.2.1 package doesn't work with the signature defined in the constructor, which no longer accepts proxy as a string.

Might just be a intermediate change for upcoming bc breaks, but figured it might be good to share this finding just in case :)

```typescript
// index.d.ts - master
class HttpsProxyAgent extends https.Agent {
    constructor(opts: HttpsProxyAgent.HttpsProxyAgentOptions)
  }

 const agent = new HttpsProxyAgent({
   host,
   port,
 }); // fails silently when supplied to ws
```

```typescript
// index.d.ts - modified
class HttpsProxyAgent extends https.Agent {
  constructor(proxy: string)
}

const agent = new HttpsProxyAgent(proxyString); // works
```

**Comment by nickelswitte** [comment]
I am not sure if this is the right place to talk about this, but using the workaround provided by @MichaelHindley, I get an error when passing the `https-proxy-agent` to `node-fetch`.

When trying to do

```typescript
fetch(url, { 
        method: 'POST', 
        headers: headers, 
        agent: agent, 
        body: data
})
```
I will get an error, that the `agent` is not of the correct type. The full error is below.

```console
(property) RequestInit.agent?: Agent | ((parsedUrl: URL) => Agent)

Type 'HttpsProxyAgent' is not assignable to type 'Agent | ((parsedUrl: URL) => Agent)'.
Property 'freeSockets' is missing in type 'HttpsProxyAgent' but required in type 'Agent'.ts(2322)

http.d.ts(361, 18): 'freeSockets' is declared here.
index.d.ts(55, 5): The expected type comes from property 'agent' which is declared here on type 'RequestInit'
```

This can be fixed using `agent: agent as any`, but does not seem like a good solution.

How to properly deal with this issue?
